pkg/plugins/resources/npm/main.go

```diff
@@ -20,10 +27,14 @@ type Spec struct {
 	Name string `yaml:",omitempty"`
 	// Defines a specific package version
 	Version string `yaml:",omitempty"`
-	// Defines registry url
+	// URL defines the registry url (defaults to `https://registry.npmjs.org/`)
 	URL string `yaml:",omitempty"`
+	// RegistryToken defines the token to use when connection to the registry
+	RegistryToken string `yaml:",omitempty"`
 	// VersionFilter provides parameters to specify version pattern and its type like regex, semver, or just latest.
 	VersionFilter version.Filter `yaml:",omitempty"`
+	// NpmrcPath defines the path to the .npmrc file
+	NpmrcPath string `yaml:"npmrcPath,omitempty"`
```

Comment:
```suggestion
	NpmrcPath string `yaml:"npmrcpath,omitempty"`
```

We decided to only use lowercase for the spec.
this is enforced by the generated jsonschema that will convert any uppercase to lowercase.